starboard/android/shared/player_components_factory.h

```diff
@@ -271,10 +276,22 @@ class PlayerComponentsFactory : public starboard::shared::starboard::player::
       constexpr int kTunnelModeAudioSessionId = -1;
       constexpr bool kForceSecurePipelineUnderTunnelMode = false;
 
-      scoped_ptr<VideoDecoder> video_decoder =
-          CreateVideoDecoder(creation_parameters, kTunnelModeAudioSessionId,
-                             kForceSecurePipelineUnderTunnelMode,
-                             max_video_input_size, error_message);
+      bool video_use_mediacodec_callback_thread = kUseMediaCodecCallbackThread;
+      if (!creation_parameters.video_mime().empty()) {
+        MimeType video_mime_type(creation_parameters.video_mime());
+        if (!video_mime_type.is_valid() ||
+            !video_mime_type.ValidateBoolParameter(
+                "mediacodeccallbackthread")) {
```

Comment:
As some browsers don't recognize MIME attributes, we should re-shape this feature to use H5VCC flags instead of mime type params, to avoid additional infrastructure issues.

We are working on the PRs of https://github.com/youtube/cobalt/pull/3186 and https://github.com/youtube/cobalt/pull/2830, and will provide a general way to set the flag via H5VCC.

As a result, this PR will be reshaped from using MIME attribute to H5VCC flags after the above PRs are merged.